File: app/src/main/java/com/iutcalendar/mainpage/ui/main/SectionsPagerAdapter.java
package com.iutcalendar.mainpage.ui.main;

import android.content.Context;
import android.util.Log;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import com.iutcalendar.event.EventFragment;
import com.iutcalendar.calendrier.Calendrier;
import com.iutcalendar.calendrier.CurrentDate;
import com.iutcalendar.data.DataGlobal;
import com.iutcalendar.data.FileGlobal;

/**
 * A [FragmentPagerAdapter] that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
public class SectionsPagerAdapter extends FragmentPagerAdapter {

    private final Context mContext;
    private final Calendrier calendrier;

    private final int countDay;

    public SectionsPagerAdapter(Context context, FragmentManager fm, Calendrier calendrier) {
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
        mContext = context;
        this.calendrier = calendrier;
        Log.d("Page", "end création SectionPagerAdapter");

        if (calendrier == null || calendrier.getFirstDay() == null) {
            this.countDay = 1;
        } else
            this.countDay = calendrier.getFirstDay().getNbrDayTo(calendrier.getLastDay()) + 1;
    }

    @Override
    public Fragment getItem(int position) {
        CurrentDate dateToLaunche;
        if (calendrier != null && calendrier.getFirstDay() != null) {
            dateToLaunche = new CurrentDate(calendrier.getFirstDay()).addDay(position);
        } else {
            dateToLaunche = new CurrentDate();
        }
        EventFragment eventFragment = new EventFragment(calendrier, dateToLaunche, FileGlobal.getFileDownload(mContext));
        return eventFragment;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return String.valueOf(DataGlobal.DAYS_OF_WEEK[position]);
    }

    @Override
    public int getCount() {
        return countDay;
    }
}

File: app/src/main/java/com/iutcalendar/mainpage/ui/main/SectionsPagerAdapter.kt
package com.iutcalendar.mainpage.ui.main

import android.util.Log
import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.iutcalendar.calendrier.CurrentDate
import com.iutcalendar.data.CachedData
import com.iutcalendar.event.EventFragment
import com.iutcalendar.mainpage.PageEventActivity


class SectionsPagerAdapter(private val parent: PageEventActivity) :
    FragmentStateAdapter(parent) {
    private var countDay = 0


    init {
        Log.d("Page", "end création SectionPagerAdapter")
        countDay =
            if (CachedData.calendrier.firstDay == null) 1 else CachedData.calendrier.firstDay!!.getNbrDayTo(CachedData.calendrier.lastDay) + 1
    }

    override fun createFragment(position: Int): Fragment {
        val dateToLaunch: CurrentDate = CachedData.calendrier.firstDay?.let { firstDay ->
            Log.d("SectionsPagerAdapter", "firstDay is  $firstDay")
            CurrentDate(firstDay).addDay(position)
        } ?: CurrentDate()
        Log.d("SectionsPagerAdapter", "get item at $dateToLaunch at position $position")
        return EventFragment(dateToLaunch)
    }


    override fun getItemCount(): Int {
        return countDay
    }
}